tritium-metrics/src/main/java/com/palantir/tritium/metrics/InstrumentedSslContext.java

```diff
@@ -44,6 +45,23 @@ public String toString() {
         return "InstrumentedSSLContext{delegate=" + context + ", name=" + name + '}';
     }
 
+    @Override
+    public boolean equals(Object other) {
+        if (this == other) {
+            return true;
+        }
+        if (other == null || getClass() != other.getClass()) {
+            return false;
+        }
+        InstrumentedSslContext that = (InstrumentedSslContext) other;
+        return context.equals(that.context) && name.equals(that.name);
+    }
+
+    @Override
+    public int hashCode() {
+        return Objects.hash(context, name);
```

Comment:
not sure how much this gets called in the pooling and maybe premature optimization, but should we avoid the varargs allocation?

Reply:
I'm not aware of anything that hashes these objects, but I can replace these with an allocation-free implementation.

Reply:
I'm fine merging as-is for now and we can adjust later if necesary